src/hg.ts

```diff
@@ -235,13 +235,13 @@ export async function exec(child: cp.ChildProcess): Promise<IExecutionResult> {
 		}),
 		new Promise<string>(c => {
 			const buffers: string[] = [];
-			on(child.stdout, 'data', b => buffers.push(b));
-			once(child.stdout, 'close', () => c(buffers.join('')));
+			on(child.stdout!, 'data', b => buffers.push(b));
```

Comment:
Let's add a check for those properties, [like the git extension does](https://github.com/microsoft/vscode/blob/314b58c8e1adb0b83f71b94f64d7e72c2f2a8259/extensions/git/src/git.ts#L185).